platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_tasks.adoc

```diff
@@ -289,5 +289,60 @@ include::{snippetsPath}/bestPractices/groupTasks-do/tests/groupTasks.good.out[]
 - <<writing_tasks.adoc#task_group_and_description,Task Group and Description>>
 
 === Tags
-
 `<<tags_reference.adoc#tag:tasks,#tasks>>`
+
+[[avoid_eager_file_collection_apis]]
+== Avoid using eager APIs on File Collections
+
+When working with Gradle's file collection types, be careful to avoid triggering dependency resolution during the configuration phase.
+
+=== Explanation
+
+Gradle's link:{javadocPath}/org/gradle/api/artifacts/Configuration.html[`Configuration`] and link:{javadocPath}/org/gradle/api/file/FileCollection.html[`FileCollection`] types extend the JDK’s `Collection<File>` interface.
+
+However, calling some available methods from this interface—such as `.size()`, `.isEmpty()`, `getFiles()`, `asPath()`, or `.toList()`—on these Gradle types will implicitly trigger resolution of their dependencies.
+Converting a `Configuration` to a simple `Set<File>`s will also cause the loss of any implicit task dependencies it carries.
+
+You should avoid using these methods when configuring your build and stick to the methods defined directly on the Gradle interfaces.
```

Comment:
Not all methods are safe, [resolve](https://docs.gradle.org/nightly/javadoc/org/gradle/api/artifacts/Configuration.html#resolve()) from your other practice). Do we want to call that out?